client/engine/engine.go

```diff
@@ -316,14 +321,31 @@ func (e *Engine) handleMessage(message protocols.Message) (EngineEvent, error) {
 		allCompleted.CompletedObjectives = append(allCompleted.CompletedObjectives, objective)
 	}
 
-	for _, voucher := range message.Vouchers() {
+	for _, entry := range message.Vouchers() {
+		// An empty objectiveId indicates a voucher sent as a payment.
+		if entry.ObjectiveId == "" {
```

Comment:
Right now we can receive a voucher in a message with or without an objective id. We rely on the convention that `entry.ObjectiveId == ""` indicates no objective for the voucher (aka a regular payment).